src/repo.rs

```diff
@@ -162,6 +163,19 @@ impl Repository {
         }
     }
 
+    /// Attempt to open an already-existing repository from a worktree.
+    pub fn open_from_worktree(worktree: &Worktree) -> Result<Repository, Error> {
+        crate::init();
```

Comment:
I think this can be skipped since the presence of `Worktree` implies it's already been called.